src/images/txb.cpp

```diff
@@ -177,18 +159,27 @@ void TXB::readHeader(Common::SeekableReadStream &txb, byte &encoding) {
 
 		_mipMaps.push_back(mipMap.release());
 
-		width      >>= 1;
-		height     >>= 1;
-		mipMapSize >>= 2;
-
-		if ((width < 1) && (height < 1))
-			break;
+		if (width  > 1) width  >>= 1;
+		if (height > 1) height >>= 1;
```

Comment:
It now loads every mipmap the texture specifies and there is only every one 1x1 texture.
This happened before for rectangular mipmaps, now it also happens for non rectangular ones too.
Besides the code is much simpler and thus hopefully easier to read.